game-core/src/main/java/games/strategy/triplea/ui/panels/map/MapPanel.java

```diff
@@ -631,16 +631,12 @@ public void drawMapImage(final Graphics g) {
           new Rectangle2D.Double(0, 0, getImageWidth(), getImageHeight());
       final Collection<Tile> tileList = tileManager.getTiles(bounds);
       for (final Tile tile : tileList) {
-        synchronized (tile.getMutex()) {
-          final Image img = tile.getImage(gameData, uiContext.getMapData());
-          if (img != null) {
-            g2d.drawImage(
-                img,
-                AffineTransform.getTranslateInstance(
-                    tile.getBounds().x - bounds.getX(), tile.getBounds().y - bounds.getY()),
-                this);
-          }
-        }
+        tile.drawImage(gameData, uiContext.getMapData());
+        final Image img = tile.getImage();
```

Comment:
Does it make sense to have `drawImage` to also return the rendered image? 

EG:
```
final var image = tile.drawImage(gameData, uiContext.getMapData())
```

Reply:
I thought about that, but by returning void it becomes obvious that this method is not just creating a new Image object, but rather altering the state of the Tile